prow/plugins/projectmanager/projectmanager.go

```diff
@@ -0,0 +1,310 @@
+	// GetIssueLabels works for PRs as they are considered issues in the API
+	var err error
+	labels, err = gc.GetIssueLabels(pe.Repo.Owner.Login, pe.Repo.Name, pe.Number)
+	if err != nil {
+		log.Infof("Error geting labels on the issue %d, ignoring processing the event", pe.Number)
+		return nil
+	}
+
+	issueURL := fmt.Sprintf("https://api.github.com/repos/%s/%s/issues/%v", pe.Repo.Owner.Login, pe.Repo.Name, pe.Number)
+	for orgRepoName, managedOrgRepo := range orgRepos {
+		for projectName, managedProject := range managedOrgRepo.Projects {
+			for _, managedColumn := range managedProject.Columns {
+				if managedColumn.Org != "" && managedColumn.Org != pe.Repo.Owner.Login {
+					continue
+				}
+				if managedColumn.State != "" && managedColumn.State != pe.PullRequest.State {
+					continue
+				}
+
+				// if labels do not match we continue to the next project
+				if !github.HasLabels(managedColumn.Labels, labels) {
+					continue
+				}
+
+				columnID := managedColumn.ID
+				// Currently this assumes columnID having a value if 0 means it is unset
+				// While it's highly unlikely that an actual project would have an ID of 0, given that
+				// these IDs are global across GitHub, this doesn't seem like an ideal solution.
+				if columnID == nil {
+					var err error
+					columnID, err = getColumnID(gc, orgRepoName, projectName, managedColumn.Name, issueURL)
+					if err != nil {
+						log.Infof("Cannot add the issue: %d to the project: %s, column: %s, error: %s", pe.Number, projectName, managedColumn.Name, err)
+						break
+					}
+				}
+				matchedColumnIDs = append(matchedColumnIDs, *columnID)
+				// if the configuration allows to match multiple columns within the same
+				// project, we will only take the first column match from the list
+				break
+			}
+		}
+	}
+	return matchedColumnIDs
+}
+
+// getColumnID returns a column id only if the issue if the project and column name provided are valid
+// and the issue is not already in the project
+func getColumnID(gc githubClient, orgRepoName, projectName, columnName, issueURL string) (*int, error) {
+	var projects []github.Project
+	var err error
+	orgRepoParts := strings.Split(orgRepoName, "/")
+	switch len(orgRepoParts) {
+	case 2:
+		projects, err = gc.GetRepoProjects(orgRepoParts[0], orgRepoParts[1])
+	case 1:
+		projects, err = gc.GetOrgProjects(orgRepoParts[0])
+	default:
+		return nil, fmt.Errorf("could not determine org or org/repo from %s", orgRepoName)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, project := range projects {
+		if project.Name == projectName {
+			columns, err := gc.GetProjectColumns(project.ID)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, column := range columns {
+				cards, err := gc.GetColumnProjectCards(column.ID)
+				if err != nil {
+					return nil, err
+				}
+
+				for _, card := range cards {
+					if card.ContentURL == issueURL {
+						return nil, errors.New(fmt.Sprintf(issueAlreadyInProject, issueURL, projectName))
```

Comment:
getColumnID either return a valid columnID or an error explaining why it could not return a valid columnID. Do you expect caller to check for nil on columnID instead of an error?